index.js

```diff
@@ -31,22 +32,40 @@ function nestedProp(obj, path) {
 	return ret;
 }
 
-function inject(string, obj, regexp) {
-	return string.replace(regexp, (match, path) => nestedProp(obj, path));
+function inject(string, obj, re) {
+	return string.replace(re, (match, path) => nestedProperty(obj, path));
 }
 
-module.exports = postcss.plugin(PLUGIN_NAME, opts => {
+const externalVars = postcss.plugin(PLUGIN_NAME, opts => {
 	opts = opts || {};
 
-	const regexp = generateRegExp(opts.prefix || '$');
+	const re = makeRe(opts.prefix || '$');
 
 	return css => {
 		css.walkDecls(decl => {
 			try {
-				decl.value = inject(decl.value, opts.data, regexp);
+				decl.value = inject(decl.value, opts.data, re);
 			} catch (err) {
 				throw decl.error(err.message, {plugin: PLUGIN_NAME});
 			}
 		});
 	};
 });
+
+externalVars.test = (str, opts) => {
+	opts = opts || {};
+	const re = makeRe(opts.prefix || '$');
+
+	return re.test(str);
+};
```

Comment:
Oh, I didn't even realize `re.test()` depends on `lastIndex`! 

Tester factory sounds reasonable :+1: It will also simplify using it as a filter.

Will make matcher factory as well.